src/Microsoft.AspNetCore.SignalR.Core/Internal/SerializedHubMessage.cs

```diff
@@ -20,8 +19,12 @@ public class SerializedHubMessage
 
         public HubMessage Message { get; }
 
-        private SerializedHubMessage()
+        public SerializedHubMessage(IEnumerable<SerializedMessage> messages)
```

Comment:
Ew IEnumerable? Who so allocatey, this is a per message allocation.

Reply:
I could make it require an array or a concrete list. This is a pubternal API though and something that lifetime managers will have to call.

Reply:
I guess we could try-cast for optimization purposes?

Reply:
Yes, we can't allocate per message like this.

Reply:
Nah IList and do  a for loop

Reply:
Derp. Don't know why I missed that one. `IReadOnlyList` ok? (assuming a for-loop).

Reply:
Does that work for array?